_posts/2020-03-02-junior-dev-advice.markdown

```diff
@@ -0,0 +1,117 @@
+---
+layout: post
+title: "Some Resources and Advice For Junior Developers"
+date: 2020-03-02 14:30:00
+categories: []
+author: rdormer
+
+---
+
+# Tough Love
+
+One of the first things that a journeyman programmer will have to
+learn when they transition from full time student to working developer
+is that things in the real world are *never* as cut and dry as their
+classes may have made it seem.  At the end of the day, software is not
+written because we love building castles of logic in the sky, it's
+written to make money.  That might seem like a trite observation, but
+it explains almost everything you observe as a working developer.
+Because perfection isn't profitable, real world applications are
+developed under less than ideal circumstances, often under intense
+pressure, and with inevitable compromises.  Large applications sprawl,
+get messy, and develop stagnant corners and scary basements.
+Questionable or just plain puzzling decisions are made by developers
+without explanation, justification, or documentation by developers who
+then leave the company, never to be seen or heard from again.  Most
+junior developers are scared of making mistakes - imagine their
+surprise when they find themselves working on something that was
+written by another junior developer who did, in fact, make some major
+mistakes - mistakes that are now their responsibility to fix.
+
+Some recommended reads at this stage:
+
+[The Number One Trait of a Great
+Developer](https://www.pixelstech.net/article/1320490159-The-Number-One-Trait-of-a-Great-Developer)
+[Every app has a scary
+basement](https://miksovsky.blogs.com/flowstate/2010/09/every-app-has-a-scary-basement.html)
+[A Taxonomy for "Bad Code
+Smells"](http://mikamantyla.eu/BadCodeSmellsTaxonomy.html)
+[The Law of Leaky
+Abstractions](https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/)
```

Comment:
@rdormer  Looks like these links are being displayed one next to the other. Can you make them look like a list?

Reply:
[](url)
<img width="1314" alt="Screen Shot 2020-03-05 at 12 46 18 PM" src="https://user-images.githubusercontent.com/7442887/76004091-59785e00-5edf-11ea-9c22-b809f9942c6c.png">